### PR TITLE
Lpal 471 configurable ecr scan

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose

Add more configuration options to ecr scan script, to cover account level ECR (as well as the management account)
The script has also been modernised to cover the newer Slack block kit and use Jinja2. 
This will hopefully give a better UX for the reports that come out of the scan

Fixes LPAL-471

## Approach

- Change script to use Jinja2 templates
- Refactor and clean up script (1st pass)
- Update Circle CI config and add scan for Lambda Containers (perfplat)

## Learning

Reusing ideas from Slack notification upgrade.

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by Raffers — APPROVED** [review]
Just a couple of comments really, but lgtm

File: scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py [hunk 1]
```diff
@@ -1,62 +1,99 @@
...
-    report_limit = ''
+    result_limit = ''
+    test_mode = False
+    slack_channel = ''
+    slack_token = ''
+    search_terms = []
+    branch = ''
+    build_url = ''
+
+    def __init__(self, args):
```
[the remark this answers is not in the document]
Comment (williamfalconeruk) [reply scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py:28]: no keyword args being used here - i'm just passing the `args` object through.

File: scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py [hunk 2]
```diff
@@ -1,62 +1,99 @@
...
             region_name='eu-west-1',
             aws_access_key_id=self.aws_iam_session['Credentials']['AccessKeyId'],
             aws_secret_access_key=self.aws_iam_session['Credentials']['SecretAccessKey'],
-            aws_session_token=self.aws_iam_session['Credentials']['SessionToken'])
-        self.images_to_check = self.get_repositories(search_term)
+            aws_session_token=self.aws_iam_session['Credentials']['SessionToken']
+            )
 
     def set_iam_role_session(self):
         if os.getenv('CI'):
```
[the remark this answers is not in the document]
Comment (williamfalconeruk) [reply scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py:71]: true - this was existing code (just moved around).

File: scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py [hunk 3]
```diff
@@ -1,62 +1,99 @@
...
             RoleSessionName='checking_ecr_image_scan',
             DurationSeconds=900
         )
         self.aws_iam_session = session
 
-    def get_repositories(self, search_term):
+    def get_repositories(self, search_terms):
         images_to_check = []
         response = self.aws_ecr_client.describe_repositories()
         for repository in response["repositories"]:
```
[the remark this answers is not in the document]
Comment (williamfalconeruk) [reply scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py:91]: proabalby better to leave as is for readability :)

[the remark this answers is not in the document]
Comment (Raffers) [reply scripts/pipeline/check_ecr_scan_results/aws_ecr_scan_results.py:91]: Absolutely, just the little fun bits with python are coming back to me & I might be getting carried away 😅

**Pull request merged**